sound/soc/sof/intel/hda.c

```diff
@@ -936,6 +936,12 @@ void hda_dsp_remove(struct snd_sof_dev *sdev)
 	/* disable DSP */
 	hda_dsp_ctrl_ppcap_enable(sdev, false);
 
+	/* Free the persistent DMA buffers used for base firmware download */
+	if (hda->cl_dmab.area)
+		snd_dma_free_pages(&hda->cl_dmab);
+	if (hda->iccmax_dmab.area)
+		snd_dma_free_pages(&hda->iccmax_dmab);
```

Comment:
I don't fully remember the iccmax details but would it be possible to use the same trick and use the same buffer as for regular firmware?

Reply:
I rather not mix the two, while iccmax afaik is not running a capture, the buffer needs to be valid before we set up the code loader DMA.
I think it would result rather confusing code to follow what is going on.